File: 월간 코드 챌린지/이진 변환 반복하기.py
def solution(s):
    # cnt = 이진 변환의 횟수
    # cnt_0 = 변환 과정에서 제거된 모든 0의 개수
    cnt, cnt_0 = 0, 0 
    while s != '1':
        cnt_0 += s.count('0')
        temp = s.count('1')
        s = bin(temp)[2:] # 이진 변환
        cnt += 1
    return [cnt, cnt_0]

File: 월간 코드 챌린지/이진 변환 반복하기.java
# https://school.programmers.co.kr/learn/courses/30/lessons/70129

import java.util.*;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        int count = 0;
        
        while (!s.equals("1")) {
            for (int i=0; i<s.length(); i++) {
                if (s.charAt(i) == '0') {
                    count++;
                }
            }
            answer[1] += count;
            s = Integer.toBinaryString(s.length() - count);
            count = 0;
            answer[0]++;
        }
        
        return answer;
        
    }
}
